pkg/controller/grouppermission/grouppermission_controller.go

```
@@ -100,54 +86,196 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
-	// Define a new Pod object
-	pod := newPodForCR(instance)
-
-	// Set GroupPermission instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
-		return reconcile.Result{}, err
+	// get list of clusterRole on k8s
+	clusterRoleList := &v1.ClusterRoleList{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, clusterRoleList)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, nil
```

This should return an error.